# PLACEMENT DATA ANALAYSIS

In [48]:
import pandas as pd
import numpy as np
import plotly
import plotly.offline as pyoff
import plotly.figure_factory as ff
from plotly.offline import init_notebook_mode, iplot, plot
import plotly.graph_objs as go

import squarify # for tree maps
%matplotlib notebook

In [143]:
#Reading the data 
p_data=pd.read_csv("Placement_Data_Full_Class.csv")
p_data.head(20)

sl_no gender  ssc_p    ssc_b  hsc_p    hsc_b     hsc_s  degree_p  \
0       1      M  67.00   Others  91.00   Others  Commerce     58.00   
1       2      M  79.33  Central  78.33   Others   Science     77.48   
2       3      M  65.00  Central  68.00  Central      Arts     64.00   
3       4      M  56.00  Central  52.00  Central   Science     52.00   
4       5      M  85.80  Central  73.60  Central  Commerce     73.30   
5       6      M  55.00   Others  49.80   Others   Science     67.25   
6       7      F  46.00   Others  49.20   Others  Commerce     79.00   
7       8      M  82.00  Central  64.00  Central   Science     66.00   
8       9      M  73.00  Central  79.00  Central  Commerce     72.00   
9      10      M  58.00  Central  70.00  Central  Commerce     61.00   
10     11      M  58.00  Central  61.00  Central  Commerce     60.00   
11     12      M  69.60  Central  68.40  Central  Commerce     78.30   
12     13      F  47.00  Central  55.00   Others   Science     65.00   
13     14      F  77.00  Central  87.00  Central  Commerce     59.00   
14     15      M  62.00  Central  47.00  Central  Commerce     50.00   
15     16      F  65.00  Central  75.00  Central  Commerce     69.00   
16     17      M  63.00  Central  66.20  Central  Commerce     65.60   
17     18      F  55.00  Central  67.00  Central  Commerce     64.00   
18     19      F  63.00  Central  66.00  Central  Commerce     64.00   
19     20      M  60.00   Others  67.00   Others      Arts     70.00   

     degree_t workex  etest_p specialisation  mba_p      status    salary  
0    Sci&Tech     No    55.00         Mkt&HR  58.80      Placed  270000.0  
1    Sci&Tech    Yes    86.50        Mkt&Fin  66.28      Placed  200000.0  
2   Comm&Mgmt     No    75.00        Mkt&Fin  57.80      Placed  250000.0  
3    Sci&Tech     No    66.00         Mkt&HR  59.43  Not Placed       NaN  
4   Comm&Mgmt     No    96.80        Mkt&Fin  55.50      Placed  425000.0  
5    Sci&Tech    Yes    55.00        Mkt&Fin  51.58  Not Placed       NaN  
6   Comm&Mgmt     No    74.28        Mkt&Fin  53.29  Not Placed       NaN  
7    Sci&Tech    Yes    67.00        Mkt&Fin  62.14      Placed  252000.0  
8   Comm&Mgmt     No    91.34        Mkt&Fin  61.29      Placed  231000.0  
9   Comm&Mgmt     No    54.00        Mkt&Fin  52.21  Not Placed       NaN  
10  Comm&Mgmt    Yes    62.00         Mkt&HR  60.85      Placed  260000.0  
11  Comm&Mgmt    Yes    60.00        Mkt&Fin  63.70      Placed  250000.0  
12  Comm&Mgmt     No    62.00         Mkt&HR  65.04  Not Placed       NaN  
13  Comm&Mgmt     No    68.00        Mkt&Fin  68.63      Placed  218000.0  
14  Comm&Mgmt     No    76.00         Mkt&HR  54.96  Not Placed       NaN  
15  Comm&Mgmt    Yes    72.00        Mkt&Fin  64.66      Placed  200000.0  
16  Comm&Mgmt    Yes    60.00        Mkt&Fin  62.54      Placed  300000.0  
17  Comm&Mgmt     No    60.00        Mkt&Fin  67.28  Not Placed       NaN  
18  Comm&Mgmt     No    68.00         Mkt&HR  64.08  Not Placed       NaN  
19  Comm&Mgmt    Yes    50.48        Mkt&Fin  77.89      Placed  236000.0

# Pre-processing

In [50]:
#Knowing about datatypes of columns
p_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 215 entries, 0 to 214
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   sl_no           215 non-null    int64  
 1   gender          215 non-null    object 
 2   ssc_p           215 non-null    float64
 3   ssc_b           215 non-null    object 
 4   hsc_p           215 non-null    float64
 5   hsc_b           215 non-null    object 
 6   hsc_s           215 non-null    object 
 7   degree_p        215 non-null    float64
 8   degree_t        215 non-null    object 
 9   workex          215 non-null    object 
 10  etest_p         215 non-null    float64
 11  specialisation  215 non-null    object 
 12  mba_p           215 non-null    float64
 13  status          215 non-null    object 
 14  salary          148 non-null    float64
dtypes: float64(6), int64(1), object(8)
memory usage: 25.3+ KB


In [51]:
#The size of data
p_data.shape

(215, 15)

In [52]:
p_data.columns

Index(['sl_no', 'gender', 'ssc_p', 'ssc_b', 'hsc_p', 'hsc_b', 'hsc_s',
       'degree_p', 'degree_t', 'workex', 'etest_p', 'specialisation', 'mba_p',
       'status', 'salary'],
      dtype='object')

In [53]:
p_data.isnull().sum()

sl_no              0
gender             0
ssc_p              0
ssc_b              0
hsc_p              0
hsc_b              0
hsc_s              0
degree_p           0
degree_t           0
workex             0
etest_p            0
specialisation     0
mba_p              0
status             0
salary            67
dtype: int64

### Salary contains 67 null values as it represnt that 67 out of 215 are not placed

In [54]:
p_data['gender'].value_counts()

M    139
F     76
Name: gender, dtype: int64

##### From above query,the given data contains 139 male students and 76 female students.

# FINDING UNIQUE VALUES IN EACH COLUMNS

In [55]:
for i in p_data.columns:
    print ("Number of unique values in {} column are {} \n The unique values are {}".format(i, len(p_data[i].unique()),p_data[i].unique()))
    print ("---------------------- \n")

Number of unique values in sl_no column are 215 
 The unique values are [  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18
  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36
  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54
  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72
  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90
  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108
 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126
 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144
 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162
 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180
 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198
 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215]
---------------------- 

Number of unique values in gen

# _Exploratory Data Analysis_

### 1)Total number of students placed

In [56]:
temp=status_data.sum()
temp

215

In [57]:
status_data=p_data['status'].value_counts()
status_data

Placed        148
Not Placed     67
Name: status, dtype: int64

#### CHART

In [58]:
trace = go.Bar(x=status_data.index,
               y= status_data.values,
               text = status_data.values,
               textposition = 'inside',
               name = 'Placement')
data = [trace]
layout = go.Layout(
    autosize=False,
    width=600,
    height=400,title = "Placement Distribution"
)

fig = go.Figure(data=data, layout=layout)
iplot(fig)

### In percentages

In [59]:
y= np.round(status_data.values/status_data.values.sum(),2)
y

array([0.69, 0.31])

In [60]:
trace = go.Bar(x=status_data.index,
               y= y,
               text = y,
               textposition = 'inside',
               name = 'Placement')
data = [trace]
layout = go.Layout(
    autosize=False,
    width=600,
    height=400,title = "Placement Distribution"
)

fig = go.Figure(data=data, layout=layout)
iplot(fig)

### The above observation shows that 69% of the students has been placed and 31% are not placed

## 2)Is there any relationship between Placement and Gender? 

In [148]:
gender_data=p_data.groupby(['gender','status']).size().reset_index().rename(columns={0:'placement count'})
gender_data

gender      status  placement count
0      F  Not Placed               28
1      F      Placed               48
2      M  Not Placed               39
3      M      Placed              100

In [146]:
gender_values=p_data[p_data['status']=='Placed']['gender'].value_counts()
gender_values

M    100
F     48
Name: gender, dtype: int64

In [128]:
gender_values[0]

100

In [130]:
gender_values[1]

48

In [108]:
trace = go.Bar(x=gender_values.index,
               y= gender_values.values,
               text = gender_values.values,
               textposition = 'inside',
               name = 'Placement')
data=[trace]
layout = go.Layout(
    autosize=False,
    width=600,
    height=400,title = "Placement Gender Distribution"
)
fig = go.Figure(data=data, layout=layout)
iplot(fig)

In [139]:
m_percentage=np.round(gender_values[0]/p_data[p_data['gender']=='M']['gender'].count(),2)*100
m_percentage

72.0

In [140]:
f_percentage=np.round(gender_values[1]/p_data[p_data['gender']=='F']['gender'].count(),2)*100
f_percentage

63.0

# Chart

In [142]:
trace = go.Bar(x=gender_values.index,
               y=[m_percentage,f_percentage],
               text = [m_percentage,f_percentage],
               textposition = 'inside',
               name = 'Placement')
data=[trace]
layout = go.Layout(
    autosize=False,
    width=600,
    height=400,title = "Placement Gender Distribution"
)
fig = go.Figure(data=data, layout=layout)
iplot(fig)

## The percentage of males is significantly high than the females.so, the males are more placed than the females.